orc8r/tools/fab/vagrant.py

```diff
@@ -12,61 +12,59 @@
 """
 import os
 import os.path
+from typing import Dict, Tuple
 
-from fabric.api import cd, env, local
+from fabric import Connection
 
 
-def __ensure_in_vagrant_dir():
+def _ensure_in_vagrant_dir(c: Connection) -> None:
     """
     Error out if there is not Vagrant instance associated with this directory
     """
-    pwd = local('pwd', capture=True)
+    pwd = c.run('pwd', hide=True).stdout.strip()
     if not os.path.isfile(pwd + '/Vagrantfile'):
         # check if we are on a vagrant subdirectory
-        with cd(pwd):
-            if not local('vagrant validate', capture=True):
-                print("Error: Vagrantfile not found")
-                exit(1)
+        if not c.run('vagrant validate', hide=True, warn=True).return_code == 0:
```

Comment:
Nit: Removing "not" and writing "...!=0" might be marginally easier to read.

Reply:
You could also write `if c.run('vagrant validate', hide=True, warn=True).return_code` for an equivalent effect, but that might be unintuitive